apidef/oas/operation.go

```diff
@@ -0,0 +1,157 @@
+package oas
+
+import (
+	"strings"
+
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+type Operations map[string]*Operation
+
+type Operation struct {
+	Allow                *Allowance `bson:"allow,omitempty" json:"allow,omitempty"`
+	Block                *Allowance `bson:"block,omitempty" json:"block,omitempty"`
+	IgnoreAuthentication *Allowance `bson:"ignoreAuthentication,omitempty" json:"ignoreAuthentication,omitempty"`
+}
+
+const (
+	allow                AllowanceType = 0
+	block                AllowanceType = 1
+	ignoreAuthentication AllowanceType = 2
+)
+
+type AllowanceType int
+
+func (s *OAS) fillPathsAndOperations(ep apidef.ExtendedPathsSet) {
```

Comment:
I'd rather do *apidef.ExtendedPathSet here so it's clear that it's fields are being modified within the functions (even if whitelist, blacklist and ignored fields are pointers)

Reply:
Actually, nothing is modified inside it